Makefile

```diff
@@ -22,11 +22,12 @@ install-dev:
 	if ! [[ -e "$(SYNCLET_DEV_IMAGE_TAG_FILE)" ]]; then echo "No dev synclet found. Run make synclet-dev."; exit 1; fi
 	./hide_tbd_warning go install -ldflags "-X 'github.com/windmilleng/tilt/internal/synclet/sidecar.SyncletTag=$$(<$(SYNCLET_DEV_IMAGE_TAG_FILE))'" ./...
 
-synclet-dev:
-	docker build -t $(SYNCLET_IMAGE):dirty -f synclet/Dockerfile .
-	docker inspect $(SYNCLET_IMAGE):dirty -f '{{.Id}}' | sed -E 's/sha256:(.{20}).*/dirty-\1/' > .synclet-dev-image-tag
-	docker tag $(SYNCLET_IMAGE):dirty $(SYNCLET_IMAGE):$$(<$(SYNCLET_DEV_IMAGE_TAG_FILE))
-	docker push $(SYNCLET_IMAGE):$$(<$(SYNCLET_DEV_IMAGE_TAG_FILE))
+synclet-dev: synclet-cache
+	docker build --build-arg baseImage=synclet-cache -t $(SYNCLET_IMAGE):dirty -f synclet/Dockerfile .
+	$(eval TAG := $(shell docker inspect $(SYNCLET_IMAGE):dirty -f '{{.Id}}' | sed -E 's/sha256:(.{20}).*/dirty-\1/'))
```

Comment:
I went with `$$(<$(SYNCLET_DEV_IMAGE_TAG_FILE))` instead of a makefile variable because, at least on mac, make runs the `$shell()` before the `docker build`, so it tries to tag the wrong image  - https://stackoverflow.com/questions/22999306/makefile-variable-assignment-executes-earl

AFAIK this is a problem on linux too, and it's just easily missable since it'll just run with the second newest image

(and fails the first time you run it since you don't have a dirty image yet)